### PR TITLE
(DOCSP-40807) Indexes > Compound indexes

### PR DESCRIPTION
**Pull request by elyse-mdb**
Add Compound Indexes to CPP Driver Indexes documentation.

Note: The existing documentation contains a section on Single-Field Indexes within the Work With Indexes page. This PR expands this section into its own page, following the page organization we use in the PHP Documentation. 

# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-40807>
Staging -
- [Optimize Queries with Indexes](https://deploy-preview-63--docs-cpp.netlify.app/indexes/)
- [Work with Indexes](https://deploy-preview-63--docs-cpp.netlify.app/work-with-indexes/)
- [Single Field Indexes](https://deploy-preview-63--docs-cpp.netlify.app/indexes/single-field-index/)
- [Compound Indexes](https://deploy-preview-63--docs-cpp.netlify.app/indexes/compound-index/)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes/compound-index.txt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   :keywords: index, query, optimization, efficiency
+
+Overview
+--------
+
+**Compound indexes** are indexes that hold references to multiple
+fields within a collection's documents. These indexes improve multi-field query and
+sort performance.
+
+To create a compound index, call the ``create_index()`` method and specify a document containing:
```
Comment (norareidy) [inline source/indexes/compound-index.txt:27]: S: We try to use full sentences when [introducing lists](https://www.mongodb.com/docs/meta/style-guide/style/lists/#introductory-text), so you can reword to:
```suggestion
To create a compound index, call the ``create_index()`` method and specify a document containing the following information: 
```

File: source/indexes/compound-index.txt [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+--------
+
+**Compound indexes** are indexes that hold references to multiple
+fields within a collection's documents. These indexes improve multi-field query and
+sort performance.
+
+To create a compound index, call the ``create_index()`` method and specify a document containing: 
+
+- The fields on which to create the index
+- The sort order for the indexed values, ``1`` for ascending or ``-1`` for descending.
```
Comment (norareidy) [inline source/indexes/compound-index.txt:30]: S: we also try to avoid [starting lists with articles](https://www.mongodb.com/docs/meta/style-guide/style/lists/#list-items)
```suggestion
- Fields on which to create the index
- Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending.
```

File: source/indexes/compound-index.txt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+.. literalinclude:: /includes/indexes/indexes.cpp
+   :start-after: start-index-compound
+   :end-before: end-index-compound
+   :language: cpp
+   :copyable:
+   :dedent:
+
+The following is an example of a query that is covered by the index
+created in the preceding code example:
```
Comment (norareidy) [inline source/indexes/compound-index.txt:63]: S: I'd reword this slightly to remove some repetition
```suggestion
The following query is covered by the index
created in the preceding code example:
```

File: source/indexes/compound-index.txt [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+Additional Information
+----------------------
+
+To learn more about compound indexes, see :manual:`Compound Indexes </core/index-compound>` in the {+mdb-server+} manual.
+
+To view runnable examples that demonstrate how to manage indexes, see
+:ref:`cpp-indexes`.
+
+To learn more about indexes in MongoDB, see the :manual:`Indexes </indexes>`
+guide in the {+mdb-server+} manual.
```
Comment (norareidy) [inline source/indexes/compound-index.txt:90]: S: you can group the Server links together. Maybe reorder like:
```suggestion
To view runnable examples that demonstrate how to manage indexes, see 
:ref:`cpp-indexes`.

To learn more about indexes, see the following resources in the {+mdb-server+} manual:

- :manual:`Indexes </indexes>`
- :manual:`Compound Indexes </core/index-compound>`
```

File: source/indexes/single-field-index.txt [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+   :values: reference
+ 
+.. meta::
+   :keywords: index, query, optimization, efficiency
+
+Overview
+--------
+
+**Single field indexes** are indexes that hold a reference to a single field of a
+document in a collection. These indexes improve single field query and
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:24]: S: This wording kinda makes it seem like the index only applies to one document. You could reword to:
```suggestion
**Single field indexes** are indexes that hold a reference to a single field of a
collection's documents. These indexes improve single field query and
```

File: source/indexes/single-field-index.txt [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+sort performance. They also support :manual:`TTL Indexes </core/index-ttl>`
+that automatically remove documents from a collection after a certain
+amount of time or at a specified clock time.
+
+To create a single-field index, call the ``create_index()`` method and specify a document containing:
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:29]: S: stay consistent with either one word (single-field) or two (single field). Other pages are inconsistent with this, but seems like most use two words.

Also, same suggestion as above about using a full sentence to introduce a list:
```suggestion
To create a single field index, call the ``create_index()`` method and specify a document containing the following information:
```

File: source/indexes/single-field-index.txt [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+- The field on which to create the index
+- The sort order for the indexed values, ``1`` for ascending or ``-1`` for descending
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:32]: S: same suggestion about articles
```suggestion
- Field on which to create the index
- Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending
```

File: source/indexes/single-field-index.txt (hunk 6 above)
Comment (norareidy) [reply source/indexes/single-field-index.txt:29]: applies to all mentions of "single-field"

File: source/indexes/single-field-index.txt [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+.. literalinclude:: /includes/indexes/indexes.cpp
+   :start-after: start-index-single
+   :end-before: end-index-single
+   :language: cpp
+   :copyable:
+   :dedent:
+
+
+The following is an example of a query that is covered by the index
+created in the preceding code example:
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:72]: S:
```suggestion
The following query is covered by the index
created in the preceding code example:
```

File: source/indexes/single-field-index.txt [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+   .. output::
+      :language: cli
+      :visible: false
+      
+      { "_id" :..., "plot" : "Peter Pan enters the nursery of the Darling children...",
+      "genres" : [ "Adventure", "Fantasy", "Family" ], "runtime" : 105,...}
+
+Additional Information
+----------------------
+
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:92]: S: same suggestion here about grouping the Server manual links

**Review by norareidy — CHANGES_REQUESTED** [review]
This looks really good! A few suggestions / comments

File: source/indexes/compound-index.txt [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+--------
+
+**Compound indexes** are indexes that hold references to multiple
+fields within a collection's documents. These indexes improve multi-field query and
+sort performance.
+
+To create a compound index, call the ``create_index()`` method and specify a document containing the following information: 
+
+- Fields on which to create the index
+- Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending.
```
Comment (norareidy) [inline source/indexes/compound-index.txt:30]: S: I'd separate the second bullet into 2 sentences. Also, since the second one has periods, [add a period](https://www.mongodb.com/docs/meta/style-guide/style/lists/#list-items) to the end of the first bullet point
```suggestion
- Fields on which to create the index.
- Sort order for the indexed values. Use ``1`` for ascending or ``-1`` for descending.
```

File: source/indexes/single-field-index.txt [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+**Single field indexes** are indexes that hold a reference to a single field in a collection's documents. 
+These indexes improve single field query and sort performance. 
+They also support :manual:`TTL Indexes </core/index-ttl>`
+that automatically remove documents from a collection after a certain
+amount of time or at a specified clock time.
+
+To create a single field index, call the ``create_index()`` method and specify a document containing the following information:
+
+- Fields on which to create the index
+- Sort order for the indexed values, ``1`` for ascending or ``-1`` for descending
```
Comment (norareidy) [inline source/indexes/single-field-index.txt:32]: S: same suggestion as Compound Index
```suggestion
- Fields on which to create the index.
- Sort order for the indexed values. Use ``1`` for ascending or ``-1`` for descending.
```

File: source/indexes.txt [hunk 12]
```diff
@@ -56,25 +58,66 @@ Single-Field Index
...
-   :end-before: end-single-field
-   :language: cpp
-   :copyable:
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/usage-examples/index-code-examples.cpp
+      :start-after: start-single-field
+      :end-before: end-single-field
+      :language: php
```
Comment (kevinAlbs) [inline source/indexes.txt:67]: Update `:language:` to `cpp`. Comment applies to other instances in this PR.

**Review by kevinAlbs — APPROVED** [review]
